server/api.go

```diff
@@ -1,22 +1,31 @@
 package server
 
 import (
+	"context"
 	"net/http"
+	"time"
 
 	"github.com/Hukyl/genesis-kma-school-entry/models"
 	"github.com/gin-gonic/gin"
 )
 
+const (
+	RatePath      = "/rate"
+	SubscribePath = "/subscribe"
+)
+
 type UserRepository interface {
 	Exists(user *models.User) (bool, error)
 	Create(user *models.User) error
 }
 
 // NewGetRateHandler is a handler that fetches the exchange rate between USD and UAH
 // from a RateFetcher interface and returns it as a JSON response.
-func NewGetRateHandler(rateFetcher RateFetcher) func(*gin.Context) {
+func NewGetRateHandler(rateService RateService, timeout time.Duration) func(*gin.Context) {
 	return func(c *gin.Context) {
-		rate, err := rateFetcher.FetchRate("USD", "UAH")
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
I'm not a Gin expert, but why don't use existing context as a base for context with timeout?
```go
ctx, cancel := context.WithTimeout(c.Request.Context(), timeout)
```